xds/src/test/java/io/grpc/xds/LoadReportClientImplTest.java

```diff
@@ -453,6 +453,86 @@ public void lrsStreamClosedAndRetried() {
     verify(backoffPolicy2, times(1)).nextBackoffNanos();
   }
 
+  @Test
+  public void lrsStreamRetryAndRereport() {
+    verify(mockLoadReportingService).streamLoadStats(lrsResponseObserverCaptor.capture());
+    StreamObserver<LoadStatsResponse> responseObserver = lrsResponseObserverCaptor.getValue();
+    assertThat(lrsRequestObservers).hasSize(1);
+    StreamObserver<LoadStatsRequest> requestObserver = lrsRequestObservers.poll();
+
+    // First LRS request sent.
+    verify(requestObserver).onNext(EXPECTED_INITIAL_REQ);
+    assertThat(logs).containsExactly("DEBUG: Initial LRS request sent: " + EXPECTED_INITIAL_REQ);
+    logs.poll();
+    assertEquals(0, fakeClock.numPendingTasks(LRS_RPC_RETRY_TASK_FILTER));
+
+    // Balancer sends back a normal response.
+    responseObserver.onNext(buildLrsResponse(100));
+    assertThat(logs).containsExactly(
```

Comment:
Verifying logs seems weird. Further more, `containsExactly()` is easily broken in the future and should not be relied on. Seems you don't need verify this.

Reply:
Those logs are valuable for tracking retries and backoffs. That's why I verifies them extensively in the other retry-related test case (`lrsStreamClosedAndRetried`), as that tests the whole bunch of retry behaviors. For this one, we concern more about load reporting back to normal after retry. Maybe it's not that critical for logs, but I would think it's not bad to keep them as there are valuable logging events involved.

Reply:
I agree logs are valuable for tracking. But verifying logs in tests is kind of cheating, it's not testing real behavior. There are multiple `containsExactly()`, so if you add any other new logs you have to change every test.

Reply:
Removed.